pybossa/jobs.py

```diff
@@ -1297,6 +1319,41 @@ def get_management_dashboard_stats(user_email):
     send_mail(mail_dict)
 
 
+def check_and_send_project_progress(project_id, conn=None):
+    from pybossa.cache.helpers import n_available_tasks
+    import pybossa.cache.users as cached_users
+    from pybossa.core import project_repo
+    from sqlalchemy.sql import text
+
+    project = project_repo.get(project_id)
+    if not project:
+        return
+
+    reminder = project.info.get('progress_reminder', {})
+    target_remaining = reminder.get("target_remaining")
+    email_already_sent = reminder.get("sent") or False
+    if target_remaining is None:
+        return
+
+    n_available_tasks = n_available_tasks(project.id)
+    if n_available_tasks > target_remaining:
+        reminder['sent'] = False
+    elif not email_already_sent:
+        # progress reached threshold and email not sent yet, send email
+        email_addr = [cached_users.get_user_email(user_id)
```

Comment:
In addition to the project owner, will this include co-owners and/or sub-admins?

Additionally, should the other email addresses be CC'd or included in the "To" field?

Reply:
Email will be sent 'TO' all project owners and co-owners, no 'CC's. And we do not provide user for entering other email addresses (use the address on their profile).